### PR TITLE
Fix/update file change

### PR DESCRIPTION
**Pull request by DKnehi**

### CODE REVIEW

File: javascript-practice/package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "parcel": "^2.12.0"
   },
   "dependencies": {
-    "commitlint": "^19.2.2"
+    "commitlint": "^19.2.2",
+    "toastify-js": "^1.12.0"
```
Comment (phuocduong-agilityio) [inline javascript-practice/package.json]: Remove this?

File: javascript-practice/src/js/views/userView.js [hunk 2]
```diff
@@ -1,19 +1,58 @@
...
+      e.preventDefault()
+      const valueEmail = document.getElementById('email').value
+      const valuePassword = document.getElementById('password').value
+      const emailErrorEl = document.getElementById('email-error')
+      const passwordErrorEl = document.getElementById('password-error')
+
+      emailErrorEl.textContent = ''
+      passwordErrorEl.textContent = ''
+
+      if (!valueEmail || !valuePassword) {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js]: Condition is complicated and duplicated, please consider to refactor

File: javascript-practice/src/js/views/userView.js [hunk 3]
```diff
@@ -1,19 +1,58 @@
...
+
+      if (!validatePassword(valuePassword)) {
+        passwordErrorEl.textContent = `${INVALID_PASSWORD}`
+
+        return
+      }
+      submitLogin(valueEmail, valuePassword)
+    })
+
+    document.getElementById('email').addEventListener('input', () => {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js]: document.getElementById('email') => Duplicated code

File: javascript-practice/src/pages/login.html [hunk 4]
```diff
@@ -1,37 +1,34 @@
...
+        </div>
+        <button class="login-button button" type="submit">Login</button>
+      </form>
+    </div>
+  </article>
+  <div id="toast-message" class="toast-message"></div> 
+  <script type="module" src="../js/main.js"></script>
+</body>
+
+</html>
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/pages/login.html]: Input empty line at the end of file

File: javascript-practice/src/styles/layouts/toast.css [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  left: 50%;
+  transform: translateX(-50%);
+  padding: 10px;
+  border-radius: 5px;
+  opacity: 0;
+  transition: opacity 0.3s ease-in-out;
+}
+
+.toast-success {
+  color: #155724;
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/styles/layouts/toast.css]: Should define variable for all color

File: javascript-practice/src/pages/login.html (hunk 4 above)
Comment (DKnehi) [reply javascript-practice/src/pages/login.html]: i done

File: javascript-practice/package.json (hunk 1 above)
Comment (DKnehi) [reply javascript-practice/package.json]: i done

File: javascript-practice/src/js/views/userView.js (hunk 2 above)
Comment (DKnehi) [reply javascript-practice/src/js/views/userView.js]: i done

File: javascript-practice/src/js/views/userView.js (hunk 3 above)
Comment (DKnehi) [reply javascript-practice/src/js/views/userView.js]: i done

File: javascript-practice/src/styles/layouts/toast.css (hunk 5 above)
Comment (DKnehi) [reply javascript-practice/src/styles/layouts/toast.css]: i done

File: javascript-practice/src/js/helpers/validation.js [hunk 6]
```diff
@@ -1,11 +1,9 @@
...
-const { REQUIRED_FIELD, INVALID_EMAIL, INVALID_PASSWORD } = ERROR_MESSAGE
+const { PASSWORD } = REGEX
 
-const { REQUIRD_FIELD,  EMAIL, PASSWORD } = REGEX
-
-export const isRequired = value => REQUIRD_FIELD.test (value !== '' ? '' : REQUIRED_FIELD);
-export const isEmail = value => EMAIL.test(value) ? '' : INVALID_EMAIL;
-export const isPassword = value => PASSWORD.test(value) ? '' : INVALID_PASSWORD;
+validatePassword = (password) => {
+  return PASSWORD.test(password)
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/helpers/validation.js]: Why don't use REGEX.PASSWORD?

File: javascript-practice/src/styles/layouts/index.css [hunk 7]
```diff
@@ -1,2 +1,4 @@
 @import "./container.css";
-@import "./header.css";
\ No newline at end of file
+@import "./header.css";
+@import "./toast.css";
+@import "./error.css";
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/styles/layouts/index.css]: Pls add empty at the end of file. Btw, I comment for issue on this. But pls check all your code for similar issue

File: javascript-practice/src/styles/layouts/toast.css [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  border-color: var( --color-boder-toast-success);
+}
+
+.toast-error {
+  color: var(--color-toast-error);
+  background-color: var(--color-background-toast-error);
+  border-color: var(--color-boder-toast-error);
+}
+
+.toast-message.show {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/styles/layouts/toast.css]: Should use nested?

File: javascript-practice/src/styles/utils/variables.css [hunk 9]
```diff
@@ -2,6 +2,13 @@
   /* start-login-page */
   --color-primary-login: var(--color-cod-gray);
   /* end-login-page */
+  --color-toast-success: #155724;
+  --color-background-toast-success: #d4edda;
+  --color-boder-toast-success: #c3e6cb;
+  --color-background-toast-error: #f8d7da;
+  --color-boder-toast-error: #f5c6cb;
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/styles/utils/variables.css]: Typo

File: javascript-practice/src/js/views/userView.js [hunk 10]
```diff
@@ -1,19 +1,59 @@
...
   bindFormLogin = (submitLogin) => {
-    this.loginFormEl.addEventListener("submit", e => {
-      e.preventDefault();
-      const valueEmail = document.getElementById('email').value;
-      const valuePassword = document.getElementById('password').value;
-      
-      submitLogin(valueEmail, valuePassword); 
-    });
+    this.loginFormEl.addEventListener('submit', (e) => {
+      e.preventDefault()
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js:22]: Check this style pls: https://github.com/airbnb/javascript?tab=readme-ov-file#semicolons

**Pull request merged**